### ISSUE TITLE
Pathfinding won't stop when a skeleton is being summoned by a necromancer which can't be seen

### ISSUE THREAD

**Issue by koisikawa** [00-Evan/shattered-pixel-dungeon#1813]
Especially when "Into Darkness" is enabled, the necromancer summons its skeleton three tiles away while the hero has a sight of only two tiles, which means that the hero won't stop pathfinding though he finds a skeleton is being summoned. And the same for the spectral necromancer.

**Comment by 00-Evan** [comment]
I'm afraid that is correct, the hero is generally only interrupted by things which occur within their field of view.

**Comment by koisikawa** [comment]
> the hero is generally only interrupted by things which occur within their field of view.

That's right, but the summoning of the skeleton actually occurs in the sight of the heroes.

Maybe the words are too abstract to understand, so I made some pictures to show that.

![Image_1725721816292](https://github.com/user-attachments/assets/0c5e9b56-ef49-4d80-80e7-03e24e66208f)

Assuming that the hero is going to the "T" tile, and a necromancer is in the way. The hero doesn't know there's a necromancer because of "Into Darkness". 

![Image_1725721820219](https://github.com/user-attachments/assets/bc0257d1-4b99-4bf9-b456-26ad2c2aff05)

In the second turn, the necromancer starts to summon its skeleton, **while the hero has already noticed the bones in front of him, but he keeps going.**

![Image_1725721822122](https://github.com/user-attachments/assets/bb3cc3b0-afda-493c-b4bf-ef004dbc2ebb)

In the third turn, because of the summoning, the hero fails to go to the next tile and is pushed back and hit by the skeleton. He finally stops.

It is a joke that every normal person would feel strange and stop when seeing some bones generated from the ground except our heroes XD.

**Comment by 00-Evan** [comment]
Oh I see, so it has nothing to do with visibility actually, it's more about the necromancer's summoning ability not interrupting the hero when it starts. Most effects with warning visuals like this do interrupt the hero, so I'll consider this.

**Comment by 00-Evan** [comment]
fixed in commit 491e478aa919d6cef71defd5659bb2546bcd62c7
